kloppy/infra/serializers/tracking/epts/reader.py

```diff
@@ -79,10 +78,10 @@ def _set_current_data_spec(idx):
             row['frame_id'] = frame_id
             row['timestamp'] = timestamp
 
-            if period_idx > len(periods):
-                if timestamp > periods[period_idx].end_timestamp:
-                    period_idx += 1
-                row['period_id'] = periods[period_idx].id
+            row['period_id'] = None
+            for i, p in enumerate(periods):
+                if timestamp >= p.start_timestamp and timestamp <= p.end_timestamp:
+                    row['period_id'] = i+1
```

Comment:
Good catch! Not sure with what file I tested the previous code but that's not working.

I think it's good to use `period.id` instead of `i + 1` to be sure it won't break when period_id's change some day.

```python
for period in periods:
   if period.start_timestamp <= timestamp <= period.end_timestamp:
     row['period_id'] = period.id
     break
```